src/server/game/Entities/Creature/Creature.h

```diff
@@ -469,7 +469,6 @@ class TC_GAME_API Creature : public Unit, public GridObject<Creature>, public Ma
 
         bool DisableReputationGain;
 
-        CreatureTemplate const* m_creatureInfo;                 // Can differ from sObjectMgr->GetCreatureTemplate(GetEntry()) in difficulty mode > 0
```

Comment:
why remove the cached pointer?

Reply:
I assumed that it was only used to identify difficulty entry https://github.com/TrinityCore/TrinityCore/blob/master/src/server/game/Entities/Creature/Creature.cpp#L507

Reply:
Fixed